.github/workflows/generate-code-owners.yml

```diff
@@ -0,0 +1,32 @@
+name: Update CODEOWNERs
+
+on:
+  schedule:
+    # https://crontab.guru/#5_8_*_*_1
+    - cron: "5 8 * * 1"
+  workflow_dispatch:
+
+jobs:
+  update:
+    runs-on: ubuntu-latest
+    if: github.repository == 'raycast/extensions'
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: "16"
+
+      - name: Generate
+        working-directory: scripts
+        run: ./generate-code-owners.js
+
+      - name: Commit
+        uses: raycast/github-actions/git-commit@master
+        with:
+          repo_dir: ${{ github.workspace }}
+          message: "Update CODEOWNERs"
```

Comment:
There should be some failure handler for this workflow in case things will go wrong, to make us aware of it.

[Here](https://github.com/raycast/extensions/blob/main/.github/workflows/generate-api-reference.yaml#L30-L42) is a ready to copy & paste code for failure Slack message.